core/src/test/java/com/graphhopper/util/PointListTest.java

```diff
@@ -110,4 +111,46 @@ public void testCopy_issue1166() {
         assertEquals(1, copy.getCapacity());
         assertEquals(9, copy.getLongitude(0), .1);
     }
+
+    @Test
+    public void testShallowCopyUnsafe() {
+        PointList pl1 = new PointList(100, true);
+        for (int i = 0; i < 1000; i++) {
+            pl1.add(i, i, 0);
+        }
+
+        PointList pl2 = pl1.shallowCopyUnsafe(100, 600);
+        assertEquals(500, pl2.size());
+        for (int i = 0; i < pl2.size(); i++) {
+            assertEquals(pl1.getLat(i + 100), pl2.getLat(i), .01);
+        }
+
+        // If you change the original PointList the shallow copy changes as well
+        pl1.set(100, 0, 0, 0);
+        assertEquals(0, pl2.getLat(0), .01);
+
+        // Create a shallow copy of the shallow copy
+        PointList pl3 = pl2.shallowCopySafe(0, 100);
+        // If we create a safe shallow copy of pl2, we have to make pl1 immutable
+        assertTrue(pl1.isImmutable());
+        assertEquals(100, pl3.size());
+        for (int i = 0; i < pl3.size(); i++) {
+            assertEquals(pl2.getLon(i), pl3.getLon(i), .01);
+        }
+
+        PointList pl4 = pl1.shallowCopyUnsafe(0, pl1.size());
+        assertTrue(pl1.equals(pl4));
```

Comment:
Is this correct? Yes it contains the same points, but it is not really equal as one is only a shallow copy.

Reply:
I would say, yes.